### PR TITLE
Add productpage with filters

### PR DESCRIPTION
**Pull request by VLeads**
Add product page with filters using useContext and useReducer
Also some code refractored

folder to check:
👉 pages/productpage
👉 context

as of now pictures are random, not in accordance with category !

netlify build is failing consistently, not able to give deploy link.
Here are screenshots

Before filter: 
ecom-storepage

-- After filter:
ecom=storepage-1

### CODE REVIEW

File: src/pages/productpage/Productpage.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      <main className="product-container">
+        <h4 className="result-heading">
+          <span className="txt-grey">
+            Enjoy shopping experience like never before
+          </span>{" "}
+          😄🛒
+        </h4>
+
+        <div className="card-container">
+          {/* {priceSortedData.map((details) => ( */}
```
Comment (foolhardy21) [inline src/pages/productpage/Productpage.jsx]: you may remove the commented code.

File: src/context/filterContext.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      : [...productList].filter((category) =>
+          filteredCategoryList.some(
+            (selectedCategory) =>
+              selectedCategory === category.categoryName.toUpperCase()
+          )
+        );
+  }
+
+  function getFilteredData(productList, state) {
+    console.log(
```
Comment (foolhardy21) [inline src/context/filterContext.js]: if not needed, this `console.log` can be removed too

**Review by foolhardy21 — APPROVED** [review]
code looks clean and compact.

File: src/components/header/Header.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .sub-nav{
     display: flex;
     justify-content: space-between;
-    padding: 0.3rem 3.5rem;
+    padding: 0rem 3.5rem;
```
Comment (MayankKumar10) [inline src/components/header/Header.css]: you don't have to write rem with 0

File: src/pages/homepage/homepage.css [hunk 4]
```diff
@@ -21,10 +21,15 @@
     left: 5.2rem;
     width: 17rem;
     height: 5rem;
+    border: 1px solid  #bf2b50
 }
 
-.hero > button:hover{
-    border: 1px solid #7575f1
+.hero > a > button:hover{
+    background-image: linear-gradient(to right, #25aae1, #4481eb, #04befe, #3f86ed);
```
Comment (MayankKumar10) [inline src/pages/homepage/homepage.css]: make it a helper function

**Review by MayankKumar10 — APPROVED** [review]
Look into these changes everything else looks good to me you can merge it

**Comment by VLeads** [comment]
Thanks for review !!

**Pull request merged**